packer/linux/scripts/install-docker.sh

```diff
@@ -6,28 +6,46 @@ DOCKER_RELEASE="stable"
 DOCKER_COMPOSE_VERSION=1.28.6
 MACHINE=$(uname -m)
 
-# This performs a manual install of Docker.
-
 # Add docker group
 sudo groupadd docker
 sudo usermod -a -G docker ec2-user
 
-# Manual install ala https://docs.docker.com/engine/installation/binaries/
-curl -Lsf -o docker.tgz "https://download.docker.com/linux/static/${DOCKER_RELEASE}/${MACHINE}/docker-${DOCKER_VERSION}.tgz"
-tar -xvzf docker.tgz
-sudo mv docker/* /usr/bin
-rm docker.tgz
+# Add the Docker yum repository, but override the url to get CentOS 7 not
+# CentOS 2 (the default with Amazon Linux 2 applied to the interpolation)
+sudo yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
+sudo yum-config-manager --setopt="docker-ce-stable.baseurl=https://download.docker.com/linux/centos/7/$MACHINE/stable" --save
+
+# Add CentOS 7 Base and Extras to the yum repos list, needed for deps for
+# packages from the Docker yum repo
+cat << 'EOF' | sudo tee /etc/yum.repos.d/centos.repo
+[base]
+name=CentOS-7 - Base
+mirrorlist=http://mirrorlist.centos.org/?release=7&arch=$basearch&repo=os
```

Comment:
Could interpolate our `$MACHINE` var instead of placing the `$basearch` literal in here, don’t feel strongly either way as the AMI isn’t likely to be rolled forward across releases or arches like a long lived install would be 🤷🏼‍♂️